src/sagemaker/tensorflow/estimator.py

```diff
@@ -185,9 +189,13 @@ def _prepare_init_params_from_job_description(cls, job_details):
             if value is not None:
                 init_params[argument] = value
 
-        framework, py_version = framework_name_from_image(init_params.pop('image'))
+        framework, py_version, tag = framework_name_from_image(init_params.pop('image'))
         init_params['py_version'] = py_version
 
+        # For backward compatibility map deprecated container tag to a framework version
+        # otherwise extract framework version from the tag itself.
```

Comment:
Could we have an utility function that would return version from the tag? Otherwise we hardcode this logic (element [0] of the tag)  in 2 different places (for both frameworks).

Reply:
i really thought about it, but the version is specific to mxnet and tensorflow and it's too little to generalize.
Also we should be able to remove it at some point in the future.

Reply:
What I had in mind is that the logic to deal with tag parsing should be externalized because we have same pattern for all the frameworks. It' same way as we do with 'framework_name_from_image'.
Version is specific to each but getting that version is common. If the pattern chanes in the future we will only update it in 1 place.

Reply:
Got it, yeah probably makes sense even for 1 line function.